File: src/app/models/Wedstrijd.ts
import { Team } from '../models/Team';

export class Wedstrijd {
  id: number;
  team1: Team;
  team2: Team;
  puntenTeam1: number;
  puntenTeam2: number;

  constructor(
    id: number,
    team1: Team,
    team2: Team,
    puntenTeam1: number = 0,
    puntenTeam2: number = 0) {
    this.id = id;
    this.team1 = team1;
    this.team2 = team2;
    this.puntenTeam1 = puntenTeam1;
    this.puntenTeam2 = puntenTeam2;
  }
}

File: backend-api/BeachPortalApi/Wedstrijd/Wedstrijd.py
from __future__ import annotations

from BeachPortalApi.Poule.Poule import Poule
from BeachPortalApi.Team.models import Team
from django.db import models


class Wedstrijd(models.Model):
    poule = models.ForeignKey(
        Poule, on_delete=models.CASCADE, related_name='wedstrijden')
    team1 = models.ForeignKey(
        Team, related_name="team1", on_delete=models.CASCADE)
    team2 = models.ForeignKey(
        Team,  related_name="team2", on_delete=models.CASCADE)
    puntenTeam1 = models.PositiveSmallIntegerField()
    puntenTeam2 = models.PositiveSmallIntegerField()

    class Meta:
        verbose_name_plural = 'Wedstrijden'

    def __str__(self):
        return f'{self.poule}: {self.team1.naam} - {self.team2.naam}'
